src/ring_theory/polynomial/hermite.lean

```diff
@@ -0,0 +1,52 @@
+/-
+Copyright (c) 2023 Luke Mantle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Luke Mantle
+-/
+
+import data.polynomial.derivative
+
+/-!
+# Hermite polynomials
+
+This file defines `hermite n`, the nth probabilist's Hermite polynomial.
+
+## Main definitions
+
+* `x_sub_dx`  : the operation `(x - d/dx)` used to recursively define the Hermite polynomials
+* `hermite n` : the nth probabilist's Hermite polynomial, defined as a `polynomial ℤ`, using
+                `x_sub_dx` recursively
+
+## References
+
+[Hermite Polynomials](https://en.wikipedia.org/wiki/Hermite_polynomials)
+
+-/
+
+noncomputable theory
+open polynomial
+
+/-- the nth probabilist's Hermite polynomial -/
+noncomputable def hermite : ℕ → polynomial ℤ
+| 0     := 1
+| (n+1) := X * (hermite n) - (hermite n).derivative
+
+@[simp] lemma hermite_succ {n : ℕ} : hermite (n+1) =  X * (hermite n) - (hermite n).derivative :=
+by rw hermite
```

Comment:
Doesn't `rfl` work here?

Reply:
When I try it, I get an error:
```
type mismatch, term
  rfl
has type
  ?m_2 = ?m_2
but is expected to have type
  hermite (n + 1) = X * hermite n - ⇑derivative (hermite n)
```

Reply:
It might have something to do with the `noncomputable` tag on the definition of `hermite n`. The code below gives the same error, but if I remove the `noncomputable` tag from the definition of `blah n`, the error goes away.
```
noncomputable def blah : ℕ → ℕ
| 0 := 0
| (n+1) := (blah n) + 1

lemma blah_succ (n : ℕ) : blah (n+1) = (blah n) + 1 := rfl
```
However, in our case, without the `noncomputable` tag on the definition of `hermite n`, I get the error
```
equation compiler failed to generate bytecode for 'hermite._main'
nested exception message:
code generation failed, VM does not have code for 'classical.choice'
```

Reply:
This is irritating but probably not worth worrying about.